.travis-ci.sh

```diff
@@ -296,13 +301,59 @@ export OCAMLRUNPARAM=b
 
     opam switch default
     opam switch remove $OPAMBSSWITCH --yes
-  else
+  elif [ "$TRAVIS_BUILD_STAGE_NAME" != "Upgrade" ]; then
     # Note: these tests require a "system" compiler and will use the one in $OPAMBSROOT
     OPAMEXTERNALSOLVER="$EXTERNAL_SOLVER" make tests ||
       (tail -n 2000 _build/default/tests/fulltest-*.log; echo "-- TESTS FAILED --"; exit 1)
   fi
+  echo -en "travis_fold:end:build\r"
 )
 
+if [ "$TRAVIS_BUILD_STAGE_NAME" = "Upgrade" ]; then
+  OPAM12DIR=~/opam1.2
+  CACHE=$OPAM12DIR/cache
+  export OPAMROOT=$CACHE/root20
+  if [[ ! -f $CACHE/bin/opam ]]; then
+    cd $OPAM12DIR
+    echo -en "travis_fold:start:opam12\r"
+    opam init --bare default git+https://github.com/ocaml/opam-repository#8be4290a
+    opam switch create opam12 --empty
+    eval `opam env`
+    OPAMEDITOR="sed -i -e 's/printconf/\"printconf\"/'" opam pin -n --edit dose 3.4.2 << EOF
+EOF
+    mkdir -p $CACHE/sources
+    cd $CACHE/sources
+    wget https://github.com/ocaml/opam/archive/1.2.2.tar.gz -O opam1.2.tar.gz
```

Comment:
Wouldn't it be OK to dl the binary here ? opam 1.2 isn't supposed to be patched anymore ?

Reply:
Why make it simple when we can make it complicated... \o/